src/main/protobuf/demo.proto

```diff
@@ -1,21 +1,29 @@
-syntax = "proto2";
+syntax = "proto3";
 
 option java_package = "com.example.protos";
 
 enum Gender {
+  UNKNOWN = 0;
   MALE = 1;
   FEMALE = 2;
 }
 
+message NullableString {
+  string value = 1;
+}
+message NullableInt {
+  int32 value = 1;
+}
+
 message Address {
-  optional string street = 1;
-  optional string city = 2;
+  NullableString street = 1;
```

Comment:
latest protoc/scalapb supports "optional string" in proto3 similar to proto2.

Reply:
Thank you for the quick reply! I was getting the following error when attempting to compile with `optional` keyword, which lead me down a rabbit hole which ended in the [article above](https://tomasbasham.dev/development/2017/09/13/protocol-buffers-and-optional-values.html) which gave some decent arguments for why an explicit `NullableType` is useful. I can understand resisting adding an opinionated way of representing optional values, though. Thoughts on how to resolve the below error? 

```
[info] Compiling 1 protobuf files to /home/mdavis/sparksql-scalapb-test/target/scala-2.12/src_managed/main
demo.proto: is a proto3 file that contains optional fields, but code generator protoc-gen-jvm_0 hasn't been updated to support optional fields in proto3. Please ask the owner of this code generator to support proto3 optional.--jvm_0_out: l-scalapb-test / Compile / protocGenerate 0s
[error] java.lang.RuntimeException: protoc returned exit code: 1
[error]         at scala.sys.package$.error(package.scala:30)
[error]         at sbtprotoc.ProtocPlugin$.compile(ProtocPlugin.scala:445)
[error]         at sbtprotoc.ProtocPlugin$.compileProto$1(ProtocPlugin.scala:633)
[error]         at sbtprotoc.ProtocPlugin$.$anonfun$sourceGeneratorTask$17(ProtocPlugin.scala:654)
[error]         at sbt.util.Difference.apply(Tracked.scala:414)
[error]         at sbt.util.Difference.apply(Tracked.scala:394)
[error]         at sbtprotoc.ProtocPlugin$.$anonfun$sourceGeneratorTask$16(ProtocPlugin.scala:643)
[error]         at sbtprotoc.ProtocPlugin$.$anonfun$sourceGeneratorTask$16$adapted(ProtocPlugin.scala:639)
[error]         at sbt.util.Tracked$.$anonfun$inputChangedW$1(Tracked.scala:219)
[error]         at sbtprotoc.ProtocPlugin$.$anonfun$sourceGeneratorTask$6(ProtocPlugin.scala:670)
[error]         at scala.Function1.$anonfun$compose$1(Function1.scala:49)
[error]         at sbt.internal.util.$tilde$greater.$anonfun$$u2219$1(TypeFunctions.scala:62)
[error]         at sbt.std.Transform$$anon$4.work(Transform.scala:68)
[error]         at sbt.Execute.$anonfun$submit$2(Execute.scala:282)
[error]         at sbt.internal.util.ErrorHandling$.wideConvert(ErrorHandling.scala:23)
[error]         at sbt.Execute.work(Execute.scala:291)
[error]         at sbt.Execute.$anonfun$submit$1(Execute.scala:282)
[error]         at sbt.ConcurrentRestrictions$$anon$4.$anonfun$submitValid$1(ConcurrentRestrictions.scala:265)
[error]         at sbt.CompletionService$$anon$2.call(CompletionService.scala:64)
[error]         at java.util.concurrent.FutureTask.run(FutureTask.java:266)
[error]         at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
[error]         at java.util.concurrent.FutureTask.run(FutureTask.java:266)
[error]         at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
[error]         at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
[error]         at java.lang.Thread.run(Thread.java:748)
[error] (Compile / protocGenerate) protoc returned exit code: 1
```